### DISCUSSION TITLE
Improving Testimonials in V3

### DISCUSSION THREAD

**Discussion by taiyab** [exercism/v3#2254] [General]
Following on from the discussion in Slack https://exercism-team.slack.com/archives/CARRG4MNC/p1600044631073800

**Andrew** 
0.106 testimonials per solution
0.226 testimonials per student

**Ricardo**
0.103 testimonials per solution
0.112 testimonials per student

**Jonah**
0.026 testimonials per solution
0.034 testimonials per student

**mjs**
0.045 testimonials per solution
0.157 testimonials per student

**neenjaw**
0.106 testimonials per solution
0.161 testimonials per student

**bobahop**
0.103 testimonials per solution
0.192 testimonials per student

**Avg. testimonials per solution: 0.081 testimonials**
**Avg. testimonials per student: 0.147 testimonials**

Even with the extremely limited dataset above, rates of testimonials per solution or student look relatively consistent even when not faceted by track popularity, student engagement, etc.

In V3, there are a few fundamental things that are changing, which should positively impact the rate of testimonials left:

- Students are only engaging in mentoring when they request it
- Students are only engaging in mentoring for Practice Exercises

There are a few other things we're considering, such as students needing to end and rate the mentor for the mentoring to be marked as "complete".

So, with that in mind, here's an open-ended question:
**How can we improve testimonials in V3, both from a mentor and student perspective?**

Whether that's to improve the rate, quality or anything else to do with them.

**Comment by SleeplessByte** [comment]
41/895 (0.046)
41/727 (0.056)

Rating 4.7

I really do think, as I said before, that it ís somewhat track dependent.

**Comment by bobahop** [comment]
I suppose I'm in favor of machinations that may increase the quantity of testimonials as long as there was no unintended consequence of lowering their quality. I suppose one way to prevent that would be to require a minumum length for a testimonial to prevent such responses as "Thanks" or "Good." But then I find myself balking at imposing more rules on a student. I really don't have a problem with the rate at which the testimonials are generated now.
